stock_lot_filter_available/models/stock_production_lot.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2021 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+
+
+class StockProductionLot(models.Model):
+    _inherit = "stock.production.lot"
+
+    product_qty = fields.Float(search="_search_product_qty")
+
+    @api.model
+    def _is_unsupported_search_operator(self, operator):
+        return operator not in ["=", "!=", ">", ">=", "<", "<="]
+
+    def _search_product_qty(self, operator, value):
+        if self._is_unsupported_search_operator(operator):
+            raise ValueError(_("Unsupported search operator"))
+        ids = []
+        domain = [
+            ("location_id.usage", "in", ["internal", "transit"]),
+            ("lot_id", "!=", False),
+        ]
+        groups = self.env["stock.quant"].read_group(
+            domain, ["lot_id", "quantity"], ["lot_id"]
+        )
+        for group in groups:
+            product_qty = group["quantity"]
+            if (
+                operator == "="
+                and product_qty == value
+                or operator == "!="
+                and product_qty != value
+                or operator == ">"
+                and product_qty > value
+                or operator == ">="
+                and product_qty >= value
+                or operator == "<"
+                and product_qty < value
+                or operator == "<="
+                and product_qty <= value
+            ):
```

Comment:
Could you use `eval()` or similar to simplify this (make it prettier and more maintainable :smiley:)?
https://docs.python.org/3/library/functions.html#eval
https://github.com/odoo/odoo/blob/13.0/odoo/tools/safe_eval.py#L307